### ISSUE TITLE
[Feature]: ORCA format reporting for KV-Cache metrics in Inference Response Header

### ISSUE THREAD

**Issue by BenjaminBraunDev** [triton-inference-server/server#7865]
### Feature Context 
As part of a larger initiative to provide advanced model server integration with load-balancers such as [LLM Instance Gateway](https://github.com/kubernetes-sigs/llm-instance-gateway), which demonstrates [dramatic performance improvements](https://docs.google.com/document/d/11ALHEF-9yOaLdbHbDjBoTY6fzejoEKiSYHzWpWqe8ZY/edit?tab=t.0), this feature would provision Triton Server with tools to easily include composite metrics in it's inference generate response header to communicate its state to the instance gateway for effective load balancing.

[This doc](https://docs.google.com/document/d/18VRJ2ufZmAwBZ2jArfvGjQGaWtsQtAP6_yF2Xn6zcms/edit?tab=t.0#heading=h.sw2xdf66jh6) captures the overall requirements for model servers to integrate with llm instance gateway. Luckily, Triton with TensorRT-LLM backend has lots of features/metrics that enable efficient load balancing such as the `nv_trt_llm_kv_cache_block_metrics` metric which we can use to derive a composite KV-cache utilization metric.

### Server Integration
The goal is to provide a way to capture [specific metrics](https://docs.google.com/document/d/18VRJ2ufZmAwBZ2jArfvGjQGaWtsQtAP6_yF2Xn6zcms/edit?tab=t.0#heading=h.a40hbpow4ioc) and report them in the ORCA format through the Triton Server frontend. To do this we will parse specific metric families from the serialized metrics provided by Triton Core libraries that Triton Server exposes at its metrics endpoint.

This feature will be controlled by a runtime environment variable `TRITON_ORCA_METRIC_FORMAT`. If not set in the environment where Triton Server is deployed, the feature is simply disabled by default. If set, `TRITON_ORCA_METRIC_FORMAT` define the format for the contents of the `endpoint-load-metrics` response header as per the [inline per-request formats](https://docs.google.com/document/d/1C1ybMmDKJIVlrbOLbywhu9iRYo4rilR-cT50OTtOFTs/edit?tab=t.0#heading=h.do9yfa1wlpk8). We plan to first implement "http" and "json" formats and later the binary protobuf format.

Therefore, currently the valid values for `TRITON_ORCA_METRIC_FORMAT` are:
- (unset)
- `"http"`
- `"json"`

Any other value will have functionality equivalent to it being unset while also logging and error.

Triton Server will report the captured metrics in that respective format in the response header. Should the desired metrics be unavailable, the server will simply not include this header in its response. Additionally, this feature will be entirely wrapped in an `#ifdef` and only be compiled if metrics are enabled on the server.

cc @yinggeh @krishung5 @jbkyang-nvi

**Comment by rmccorm4** [comment]
Hi @BenjaminBraunDev, can this issue be closed in favor of the ongoing PR: https://github.com/triton-inference-server/server/pull/7839?

**Comment by BenjaminBraunDev** [comment]
> Hi [@BenjaminBraunDev](https://github.com/BenjaminBraunDev), can this issue be closed in favor of the ongoing PR: [#7839](https://github.com/triton-inference-server/server/pull/7839)?

Yes, I just want to make sure it's still accessible as it has more background details on the feature than the PR.
